tools/android_lint/bin/main.dart

```diff
@@ -154,11 +159,34 @@ ArgParser setupOptions() {
   return argParser;
 }
 
+/// On macOS, we can try to find Java 1.8.
+///
+/// Otherwise, default to whatever JAVA_HOME is already.
+Future<String> getJavaHome() async {
+  if (Platform.isMacOS) {
+    final ProcessResult result = await processManager.run(
+      <String>['/usr/libexec/java_home', '-v', '1.8'],
```

Comment:
I think we want an exit code check here too. This may fail with "No such file or directory" in some edge cases, like if there's no JDK install at all.

Reply:
I'm checking earlier if this works, but should check here too. Done.